python/mxnet/io.py

```diff
@@ -141,12 +141,23 @@ def __str__(self):
             label_shapes)
 
 class DataIter(object):
-    """The base class of a data iterator.
+    """The base class for an MXNet data iterator.
+
+    All I/O in MXNet goes through children of this class. Data-iterators in
```

Comment:
goes through children -> is handled by specializations (to avoid tree a analogy)